### PR TITLE
Cross-platform support

### PR DESCRIPTION
**Pull request by poeschko**
This implements the idea described here: https://medium.com/@g_syner/for-the-most-part-i-really-like-your-solution-664c8248ec30#.4ekcegbww

Note that I also changed the way `npm` is called. Instead of going through `npmi`, it's now executing `npm` directly. This was necessary to call `npm rebuild`. Also, npm's maintainers say that one shouldn't rely on any programmatic interface to npm (other than the command-line interface).

Let me know how you like these changes and if there's anything I can improve.

### CODE REVIEW

File: src/checkout-node-modules.es6 [hunk 1]
```diff
@@ -1,28 +1,30 @@
...
+    let leaveAsIs = false;
     log.setLevel(verbose ? `debug`: `info`);
     return readFilePromise(`${cwd}/package.json`, `utf-8`)
     .then((packageJsonContent) => {
         // replace / in hash with _ because git does not allow leading / in tags
+        let packageJson = JSON.parse(packageJsonContent);
         packageJsonSha1 = crypto.createHash(`sha1`).update(packageJsonContent).digest(`base64`).replace(/\//g, "_");
-        packageJsonVersion = packageJsonContent.version;
-        log.debug(`Sha1 of package.json is ${packageJsonSha1}`);
+        packageJsonVersion = packageJson.version;
```
Comment (bestander) [inline src/checkout-node-modules.es6:26]: good find!

**Comment by bestander** [comment]
Hey, @poeschko, after the merge CI throws errors when running tests https://codeship.com/projects/75106/builds/11121475
Could you check what happened?

**Comment by bestander** [comment]
Too bad CI did not run for the pull request, I thought Codeship can do that.

**Comment by poeschko** [comment]
Well, that's a bummer... Tests were running fine for me locally. I can't see the CI output (it just redirects to the codeship.com landing page) – is it private by any chance? Definitely want to sort this out.

**Comment by bestander** [comment]
Yeah, looks like it is a private project.
Can't find how to make it public.
Let me try some other CI system

**Comment by bestander** [comment]
Good to be in 2016 :)
Another CI is configured, https://circleci.com/gh/bestander/npm-git-lock/tree/circleci.
If you need to switch node from 0.12, see circle.yml file.

Please reopen the the PR, the CI should pick it up now

**Pull request merged**